src/modules/sub-items/components/table-view/table.view.component.js

```diff
@@ -93,10 +182,42 @@ export default class TableViewComponent extends Component {
                 generateLink={generateLink}
                 onItemSelect={onItemSelect}
                 isSelected={isSelected}
+                columnsVisibility={columnsVisibility}
             />
         );
     }
 
+    renderColumnsToggler() {
+        if (!this.state.isColumnsTogglerOpen) {
+            return null;
+        }
+
+        const { columnsVisibility } = this.state;
+
+        return (
+            <div className="c-table-view__columns-toggler-panel">
+                <ul className="c-table-view__columns-toggler-list">
+                    {Object.keys(columnsVisibility).map((columnKey) => {
+                        const isColumnVisible = columnsVisibility[columnKey];
+                        const label = this.headerLabels[columnKey];
+
+                        return (
+                            <li
```

Comment:
MAybe you should create a separate component so you could avoid re-renders caused by binding params to function in `onClick`. What do you think?

Reply:
Yes, I was thinking about extracting whole toggler panel to separate component, do you line this idea?